core/primitives-core/src/types.rs

```diff
@@ -23,7 +29,77 @@ pub type ShardId = u64;
 /// Balance is type for storing amounts of tokens.
 pub type Balance = u128;
 /// Gas is a type for storing amount of gas.
-pub type Gas = u64;
+#[derive(
+    Debug,
+    Default,
+    BorshSerialize,
+    BorshDeserialize,
+    Serialize,
+    Deserialize,
+    Clone,
+    Copy,
+    Hash,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+)]
+pub struct Gas(u64);
+
+impl Gas {
+    pub fn new(val: u64) -> Self {
+        Self(val)
+    }
```

Comment:
Consider `impl From<u64> for Gas`

Reply:
Good idea!  I suppose that will also allow me to get rid of `get()`?  Also do you have any rule of thumbs or links that can help one decide when to use `new` vs `impl From`

Reply:
I found https://www.reddit.com/r/rust/comments/k7z66j/should_i_define_new_or_implement_from_for/.

Reply:
>  Also do you have any rule of thumbs or links that can help one decide when to use new vs impl From

For newtypes that represent dimensional data (i.e. units) from/into my intuition is that this is more similar to conversion from `u32` to `u64`, than anything else which have plenty of rules of thumbs and conventions built around `From`.

OTOH, there are plenty of cases where `From`/`Into` is just impossible to use sensibly (e.g. the type needs two values to construct it…) too, so an inherent constructor method is the way to go there.

Usually I just go with the gut though.